### PR TITLE
Fix crash when loading geo data

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

**Comment by etiennehomer** [comment]
The crash is still present. This fix is relevant but a checkout needs to be done in the back

File: src/components/network-map-tab.js [hunk 1]
```diff
@@ -422,56 +423,73 @@ export const NetworkMapTab = ({
...
                     console.error(error.message);
-                    setErrorMessage(
-                        intlRef.current.formatMessage(
-                            { id: 'geoDataLoadingFail' },
-                            { studyUuid: studyUuid }
-                        )
-                    );
+
+                    // we display the error to the user only if the node is built
+                    if (isNodeBuilt(currentNodeRef.current)) {
```
Comment (etiennehomer) [inline src/components/network-map-tab.js:485]: You could use isSameNodeAndBuilt() here nope ?

File: src/components/network-map-tab.js [hunk 2]
```diff
@@ -528,12 +546,16 @@ export const NetworkMapTab = ({
...
                 console.error(error.message);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
-                );
+
+                // we display the error to the user only if the node is built
+                if (isNodeBuilt(currentNodeRef.current)) {
```
Comment (etiennehomer) [inline src/components/network-map-tab.js:551]: Same

File: src/components/network-map-tab.js [hunk 3]
```diff
@@ -522,18 +549,29 @@ export const NetworkMapTab = ({
...
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
-                );
+
+                // we display the error to the user only if the node is built
+                if (
+                    isSameNodeAndBuilt(currentNodeRef.current, nodeBeforeFetch)
```
Comment (sBouzols) [inline src/components/network-map-tab.js:562]: Add this `isSameNodeAndBuilt` check in the `fetchLinePositions.then` and `fetchSubstationPositions.then` as well

File: src/components/network-map-tab.js [hunk 4]
```diff
@@ -422,56 +423,73 @@ export const NetworkMapTab = ({
...
-                            (!lineFullPath && substationsDataChanged)
-                                ? new Map(geoDataRef.current.linePositionsById)
-                                : geoDataRef.current.linePositionsById
-                        );
-                        newGeoData.updateSubstationPositions(
-                            notFoundSubstationIds,
-                            fetchedSubstationPositions
-                        );
-                        newGeoData.updateLinePositions(
+                    if (
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/network-map-tab.js:430]: I second that way of laying the code :)

**Review by etiennehomer — APPROVED** [review]

File: src/components/network-map-tab.js (hunk 3 above)
Comment (sBouzols) [reply src/components/network-map-tab.js]: done

**Review by sBouzols — APPROVED** [review]
Code Review OK
Test OK
Console Warning Check OK

**Pull request merged**